wagtail/admin/templates/wagtailadmin/shared/menu_item.html

```diff
@@ -1,3 +1,12 @@
+{% load wagtailadmin_tags %}
+
 <li class="menu-item{% if active %} menu-active{% endif %}">
-    <a href="{{ url }}" class="{{ classnames }}"{{ attr_string }}>{{ label }}</a>
+    <a href="{{ url }}"
+    {% if not icon_name %}
+       class="{{ classnames }}"
+    {% endif %}
```

Comment:
I think we need to discuss the backwards-compatibility consequences of this.

Reply:
Added.

Reply:
> I think we need to discuss the backwards-compatibility consequences of this. Do we need to stop passing classnames if `icon_name` is set?

I think we should continue passing `classnames` for the reasons I mentioned in https://github.com/wagtail/wagtail/pull/4821/files#r307966908

Despite having written that, last summer, I had the code to fix this template to still accept `classnames` no matter what stashed and uncommitted. I will add it to the PR momentarily.